src/signals_notebook/entities/parallel_experiment/sub_experiment.py

```diff
@@ -105,3 +108,65 @@ def get_html(self) -> str:
         log.info('Html template for %s:%s has been rendered.', self.__class__.__name__, self.eid)
 
         return template.render(data=data)
+
+    @classmethod
+    def load(cls, path: str, fs_handler: FSHandler, parallel_experiment: ParallelExperiment) -> None:
+        from signals_notebook.item_mapper import ItemMapper
+
+        metadata = json.loads(fs_handler.read(fs_handler.join_path(path, 'metadata.json')))
+        sub_experiment = cls.create(
+            parallel_experiment=parallel_experiment, description=metadata['description'], force=True
+        )
+        existing_chemical_drawing = [i for i in sub_experiment.get_children()
+                                     if i.type == EntityType.CHEMICAL_DRAWING][0]
+
+        child_entities_folders = fs_handler.list_subfolders(path)
+        for child_entity in child_entities_folders:
+            child_path = fs_handler.join_path(path, child_entity)
+            metadata = json.loads(fs_handler.read(fs_handler.join_path(child_path, 'metadata.json')))
+            child_entity_type = child_entity.split(':')[0]
+            entity_type = ItemMapper.get_item_class(child_entity_type)
+
+            if child_entity_type == EntityType.CHEMICAL_DRAWING:
+
+                if not metadata.get('reactants') or not metadata.get('products'):
+                    continue
+
+                reactants = [
+                    Structure(
+                        id=reactant['id'],
+                        type=ChemicalStructure.REACTANT,
+                        inchi=reactant['inchi'],
+                        cdxml=reactant['cdxml'],
+                    )
+                    for reactant in metadata['reactants']
+                ]
+
+                products = [
+                    Structure(
+                        id=product['id'], type=ChemicalStructure.PRODUCT, inchi=product['inchi'], cdxml=product['cdxml']
+                    )
+                    for product in metadata['products']
+                ]
+                reagents = [
+                    Structure(
+                        id=reagent['id'], type=ChemicalStructure.REAGENT, inchi=reagent['inchi'], cdxml=reagent['cdxml']
+                    )
+                    for reagent in metadata['reagents']
+                ]
+
+                for reactant in reactants:
```

Comment:
maybe structurize code by actions?
U have so bad and long method. I cannot catch the idea after 30 code lines